quipucords/fingerprinter/__init__.py

```diff
@@ -297,7 +293,11 @@ def _process_source(report_id, source):
 
         if fingerprint is not None:
             fingerprint['report_id'] = report_id
-            fingerprint[SOURCES_KEY] = [source_id]
+            fingerprint[SOURCES_KEY] = \
+                {source_name: {'source_type': source_type,
+                               'name': source_name}}
+            fingerprint['sources_info'] = [{'source_type': source_type,
```

Comment:
Let's convert that to a dictionary merge --> https://github.com/quipucords/quipucords/blob/master/quipucords/fingerprinter/__init__.py#L539-L542